### PR TITLE
feat: update ESLint, syntax

### PR DESCRIPTION
**Pull request by tehhowch**
Optional Chaining & nullish coalescing are further supported, to the point where we can be comfortable using them.

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -311,11 +311,8 @@
      * @param {string} url The URL that invoked the function call.
      */
     function recordCrowns(settings, xhr, url) {
-        // TODO: Replace with optional chaining, once full spec compliance is obtained.
-        if (!xhr.responseJSON || !xhr.responseJSON.page || !xhr.responseJSON.page.tabs
-            || !xhr.responseJSON.page.tabs.kings_crowns || !Array.isArray(xhr.responseJSON.page.tabs.kings_crowns.subtabs)
-            || !xhr.responseJSON.page.tabs.kings_crowns.subtabs[0] || !xhr.responseJSON.page.tabs.kings_crowns.subtabs[0].mouse_crowns
-        ) {
+        const mouseCrowns = xhr.responseJSON?.page?.tabs?.kings_crowns?.subtabs?.[0]?.mouse_crowns;
```
Comment (DevJackSmith) [inline src/scripts/main.js:314]: this is great! didn't know this notation existed!

Comment (DevJackSmith) [inline src/scripts/main.js:314]: this is great! didn't know this notation existed!

**Review by DevJackSmith — APPROVED** [review]
lgtm. I'll give a day or two for others to review if they want. Thanks!

**Review by AardWolf — COMMENTED** [review]
I appreciate being able to use the hook-dot and also some of this simplification of objects. Thank you.

**Pull request merged**